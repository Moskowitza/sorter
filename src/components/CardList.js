import React from 'react';
import PropTypes from 'prop-types';
import Card from './Card';

const CardList = ({ robots }) => (
  <div>
    {
        robots.map((user, i) => (
          <Card
            key={robots[i].id}
            id={robots[i].id}
            name={robots[i].name}
            email={robots[i].email}
          />
        ))
      }
  </div>

);
CardList.defaultProps = {
  robots: [],
};
CardList.propTypes = {
  robots: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
    id: PropTypes.number.isRequired,
  })),

};
export default CardList;
