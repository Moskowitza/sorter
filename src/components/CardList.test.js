import { shallow, mount, render } from 'enzyme';
import React from 'react';

import CardList from './CardList';


test('cardList snapshot', () => {
  const mockRobots = [{
    id: 1,
    name: 'fred',
    email: 'fred@email.com',
  },
  {
    id: 2,
    name: 'fred',
    email: 'fred@email.com',
  },
  ];
  // eslint-disable-next-line no-undef
  const cardList = shallow(<CardList robots={mockRobots} />);
  expect(cardList).toMatchSnapshot();
});
