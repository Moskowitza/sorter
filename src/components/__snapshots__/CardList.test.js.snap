// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardList snapshot 1`] = `ShallowWrapper {}`;
